Extend Apps docuentation (#131) (#160)
mirumee/saleor-docs: docs/developer/extending/apps.mdx

##### Schedule installation

<Chart
  definition={`
  sequenceDiagram
    Client ->>+ Saleor: mutation appInstall
    Saleor ->> Saleor: Validate recieved data
    Saleor ->>+ Background Worker: Schedule installation
    Background Worker -->>- Saleor: Scheduled
    Saleor -->>- Client: return appInstallation
`}
/>

##### Install an App

<Chart
  definition={`
  sequenceDiagram
    participant Saleor
    participant DB
    participant App
    Saleor ->>+ App: GET /path/to/manifest.json
    App -->>- Saleor: App manifest
    Saleor ->> Saleor: Validate manifest
    Saleor ->>+ DB: Create App 
    DB -->>- Saleor: App object
    Saleor ->> Saleor: Generate new auth token for app
    Saleor ->>+ App: POST /token/target/url/<br>{"auth_token": "new-app-token"}
    App -->>- Saleor: HTTP 200 OK
`}
/>

#### Retrieving ongoing installations

To fetch all ongoing and failed installations, use the `appsInstallations` query. 
This query is only available to staff users with the `MANAGE_APPS` permission. 
The query returns list of `AppInstallation` objects.

```graphql
{
  appsInstallations{
    id
    status
    appName
  }
}
```

> The query returns simillar response:

Reviewer: ```suggestion
The query returns a similar response:
```

> ```graphql {2-4}

Reviewer 2: You will need to update the highlighted range as well.

}
```

### Retrieving all Apps

To fetch list of all apps, use `apps` query. This query is only available to staff users with the `MANAGE_APPS` permission.

```graphql
{
  apps(first: 5){
    edges{
      node{
        id
        permissions{
          code
        }
        name
        isActive
        webhooks{
          name
        }
        
      }
    }
  }
}
```

> The query returns simillar response:

Reviewer: ```suggestion
The query returns a similar response:
```

```json
{
  "data": {
    "apps": {
      "edges": [
        {
          "node": {
            "id": "QXBwOjE=",
            "permissions": [
              {
                "code": "MANAGE_ORDERS"
              }
            ],
            "name": "Order App",
            "isActive": true,
            "webhooks": []
          }
        }
      ]
    }
  }
}
```

### Activating an App
To activate an app, use the `appActivate` mutation. This mutation is only available to staff users with the `MANAGE_APPS` permission. 
Staff user can't activate app which has more permissions than he has for his own. 

The mutation takes the following fields:
- `id`: the id of the `App`, required.

```graphql
mutation {
  appActivate(id:"QXBwOjE="){
    app{
      id
      isActive
    }
    appErrors{
      field
      message
      code
    }
  }
}
```

> The query returns simillar response:

Reviewer: ```suggestion
The query returns a similar response:
```

```json
{
  "data": {
    "appActivate": {
      "app": {
        "id": "QXBwOjE=",
        "isActive": true
      },
      "appErrors": []
    }
  }
}
```
### Deactivating an App

To deactivate an app, use the `appDeactivate` mutation. This mutation is only available to staff users with the `MANAGE_APPS` permission. 

> Staff user can't appDeactivate app which has more permissions than he has for his own. 

Reviewer: ```suggestion
Staff user can't deactivate an app which has more permissions than he has for his own. 
```

Reviewer: Also, to not assume gender ("he") you can say "Staff users [...] they have for their own" :)

Reviewer 2: Yes, always use the neutral "they"/"their" instead of "he"/"him"/"his".

:::note

> The deactivating app will disable all auth-tokens generated for an app. 

Reviewer: ```suggestion
Deactivating an app will disable all auth-tokens generated for an app. 
```

The app won't receive a webhook payload.
:::

The mutation takes the following fields:
- `id`: the id of the `App`, required.

```graphql
mutation {
  appDeactivate(id:"QXBwOjE="){
    app{
      id
      isActive
    }
    appErrors{
      field
      message
      code
    }
  }
}
```

> The query returns simillar response:

Reviewer: ```suggestion
The query returns a similar response:
```
